File: BTS/C++/polymorphismeCours/BarreRonde.cpp
#include "BarreRonde.h"
#include "Barre.h"
#include <math.h>

BarreRonde::BarreRonde(const float _densite, const float _longueur, const string _referenceBarre, const float _rayon)
    : Barre(_densite, _longueur, _referenceBarre), rayon(_rayon)
{
    cout << "Je suis le constructeur a argument de BarreRonde." << endl;
}

BarreRonde::BarreRonde()
    : Barre(), rayon(2)
{
    cout << "Je suis le constructeur par defaut de BarreRonde." << endl;
}

BarreRonde::BarreRonde(const BarreRonde &barreRonde)
{
    cout << "Je suis le constructeur par copie de BarreRonde." << endl;
}

BarreRonde::~BarreRonde()
{
    cout << "BarreRonde detruite." << endl;
}

float BarreRonde::calculerSection()
{
    return M_PI*rayon*rayon;
}

float BarreRonde::calculerMasse()
{
    return longueur*densite*calculerSection();
}

File: BTS/Java/Premiere annee/fonderie/BarreRonde.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package fonderie;

/**
 *
 * @author therbron
 */
public class BarreRonde extends Barre{
    
    private double rayon;

    public BarreRonde() {
        super();
        this.rayon = 3;
        System.out.println("Constructeur defaut BarreRonde");
    }

    public BarreRonde(double rayon, double densite, double longueur, String reference) {
        super(densite, longueur, reference);
        this.rayon = rayon;
        System.out.println("Constructeur parametre et super parametres BarreRonde");
    }

    public BarreRonde(BarreRonde autreBarre) {
        super(autreBarre);
        this.rayon = autreBarre.rayon;
        System.out.println("Constructeur copie BarreRonde");
    }
    
    double calculerMasse()
    {
        return longueur*densite*(2*Math.PI*rayon);
    }
    
}
